syntax = "proto3";
package CARTAVIS;

service CartaBackend {
    // Calls an action on the CARTA frontend
    rpc CallAction (ActionRequest) returns (ActionReply) {}
}

// A request with the action path, name and parameters, and a flag to make the call asynchronous
message ActionRequest {
    sfixed32 session_id = 1;
    string path = 2;
    string action = 3;
    string parameters = 4;
    bool async = 5;
}

// A response with the success status resulting from the action call, and a possible error message
message ActionReply {
    bool success = 1;
    string message = 2;
    string response = 3;
}
